src/Namicorn.ts

```diff
@@ -78,63 +79,110 @@ class Namicorn {
     }
   }
 
-/**
- * Resolves the given domain
- * @async
- * @param {string} domain - domain name to be resolved 
- * @returns {Promise<NamicornResolution>} - Returns a promise that resolves in an object 
- */
+  /**
+   * Resolves the given domain
+   * @async
+   * @param {string} domain - domain name to be resolved
+   * @returns {Promise<NamicornResolution>} - Returns a promise that resolves in an object
+   */
   async resolve(domain: string): Promise<NamicornResolution> {
-    if (this.blockchain) {
-      return await this.resolveUsingBlockchain(domain);
-    } else {
-      const response = isNode()
-        ? await fetch(`${this.api}/${domain}`)
-        : await window.fetch(`${this.api}/${domain}`);
-      return response.json();
+    try {
+      if (this.blockchain) {
+        return await this.resolveUsingBlockchain(domain);
+      } else {
+        const response = isNode()
+          ? await fetch(`${this.api}/${domain}`)
+          : await window.fetch(`${this.api}/${domain}`);
+        return response.json();
+      }
+    } catch (err) {
+      throw err;
     }
   }
 
-/**
- * Resolves give domain name to a specific currency address if exists
- * @async
- * @param {string} domain - domain name to be resolved
- * @param {string} currencyTicker - currency ticker like BTC, ETH, ZIL
- * @returns {Promise<string>} - A promise that resolves in an address or null
- */
-  async address(domain: string, currencyTicker: string): Promise<string> {
-    const data = await this.resolve(domain);
+  /**
+   * Resolves give domain name to a specific currency address if exists
+   * @async
+   * @param {string} domain - domain name to be resolved
+   * @param {string} currencyTicker - currency ticker like BTC, ETH, ZIL
+   * @returns {Promise<string>} - A promise that resolves in an address or null
+   */
+  async address(
+    domain: string,
+    currencyTicker: string,
+  ): Promise<string | null> {
+    try {
+      return await this.addressOrThrow(domain, currencyTicker);
+    } catch (error) {
+      if (error instanceof ResolutionError) {
+        return null;
+      } else {
+        throw error;
+      }
+    }
+  }
+
+  /**
+   * Resolves given domain to a specific currency address or throws
+   * @param domain - domain name
+   * @param currencyTicker - currency ticker such as
+   *  - ZIL
+   *  - BTC
+   *  - ETH
```

Comment:
I am not sure this list will be formatted correctly in docs. Use single line if it is not.